docs/fml-front-end-format.md

```diff
@@ -0,0 +1,59 @@
+---
+id: fml-front-end-format
+title: Feature Manifest Language Front-end Format
+slug: /fml-front-end-format
+---
+
+- Status: new
+- Deciders: @travis79, @jhugman, @teshaq, @k88hudson
+- Date: 2021-10-26
+
+## Context and problem statement
+
+In order to allow product teams to define experimentable application features, there needs to be a manifest file which defines these in a way that is understandable to the Nimbus ecosystem.  The feature manifest should define data types used in the application code to configure features. It should also be able to define a complete default configuration for each application feature.
+
+## Decision drivers
+
+- It should not be onerous to write and maintain for a junior engineer or intern
+- A single feature should be readable/guessable by a product owner or designer
+- Once written, it should be easy to navigate and read, even if there are many features
+- It should map easily to Kotlin, Swift (and Rust)
+- No algebraic types
+- No inheritance / polymorphism
+- Defaults should be specifiable at the property level
+- Defaults should be specifiable at the feature level
```

Comment:
Would alignment with desktop be another decision driver? In the hopes that maybe we can (in the future at least) have one format/schema for both?

Reply:
Yes, good call! Currently, there isn't anything in the desktop schema that couldn't be represented but I would like to have the same schema eventually.